src/plugins/badge.coffee

```diff
@@ -0,0 +1,583 @@
+      when 'embedded' then @_renderEmbedded()
+
+  $floatingBadgePlugin: -> @parent_doc.getElementById('sa-badge-floating-plugin')
+  $embeddedBadgePlugin: -> @parent_doc.getElementById('sa-badge-embedded-plugin')
+  $embeddedBadgeMoreButton: -> @parent_doc.getElementById('sa-badge-embedded-more-button')
+  $closeModalButton: -> @parent_doc.getElementById('sa-badge-modal-close-button')
+  $showReviewsButton: ->
+    switch context().configuration.display
+      when 'floating' then @$floatingBadgePlugin()
+      when 'embedded' then @$embeddedBadgeMoreButton()
+
+  ###
+  Adds the required css for the plugin
+  ###
+  addStyle: ->
+    style = document.createElement('style')
+    style.id = 'sa-badge-style'
+    style.type = 'text/css'
+
+    # IE compatibility
+    if style.styleSheet
+      style.styleSheet.cssText = STYLE(context().configuration)
+    else
+      style.appendChild(document.createTextNode(STYLE(context().configuration)))
+
+    @_head().appendChild(style)
+
+  _renderFloating: ->
+    $el = document.createElement('div')
+    $el.id = 'sa-badge-floating-plugin'
+    $el.innerHTML = FLOATING_TEMPLATE(stars: @_ratingToStars(context().data.rating))
+    @parent_doc.body.appendChild($el)
+
+  _renderEmbedded: ->
+    rating = context().data.rating
+    @$embeddedBadgePlugin().innerHTML = EMBEDDED_TEMPLATE(rating: rating.toFixed(1), stars: @_ratingToStars(rating))
+
+  _bindHandlers: ->
+    @_attachEvent(@$showReviewsButton(), 'click', @_onClickShowReviewsButton)
+
+  _attachEvent: (element, event, handler) ->
+    return element.addEventListener event, handler, false  if element.addEventListener
+    element.attachEvent "on#{event}", handler              if element.attachEvent
+
+  _createModal: ->
+    @$modal = document.createElement('div')
+    @$modal.id = 'sa-badge-modal'
+    @$modal.innerHTML = MODAL_INNER_TEMPLATE(src: @_iframeSrc())
+    @parent_doc.body.appendChild(@$modal)
+    @_attachEvent(@$closeModalButton(), 'click', @_onClickModalCloseButton)
+    @_attachEvent(@$modal, 'click', @_onClickModal)
+    @_attachEvent(@parent_doc, 'keyup', @_onKeyUp)
+
+  _destroyModal: ->
+    return unless @$modal?
+
+    @$modal.parentNode.removeChild(@$modal)
+    @$modal = null
+
+  _showModal: ->
+    return @_createModal() unless @$modal?
+
+    @$modal.style.display = 'block'
+
+  _hideModal: ->
+    return unless @$modal?
+
+    @$modal.style.display = 'none'
+
+  _onClickShowReviewsButton: => @_showModal()
+  _onClickModal: => @_hideModal()
+  _onClickModalCloseButton: => @_hideModal()
+
+  _onKeyUp: (event) =>
+    if (event.keyCode || event.which) == 27 # ESC
+      @_hideModal()
+
+      # Destroy modal in order to reload iframe on next _showModal,
+      # as browsers interrupt iframe loading on ESC.
+      @_destroyModal()
+
+  _head: -> @parent_doc.head || @parent_doc.getElementsByTagName('head')[0]
+
+  _setParentDoc: -> @parent_doc = window.parent.document
+
+  _ratingToStars: (rating = 0, limit = 5) ->
```

Comment:
Since we get the rating from `context`, we don't need to pass it a method argument.
This way we would simplify the method calls [here](https://github.com/skroutz/analytics.js/pull/32/commits/838601d6d635d983481d0899857712bf77137279#diff-0b4af4b95323cafaa714c419ffd8c491R503) and [here](https://github.com/skroutz/analytics.js/pull/32/commits/838601d6d635d983481d0899857712bf77137279#diff-0b4af4b95323cafaa714c419ffd8c491R508).